football_dictionaries/assignment_3.py

```diff
@@ -1,2 +1,186 @@
+    if br_goalkeepers == []:
+        pass
+    else:
+        bra_dict['GK'] = br_goalkeepers
+    bra_dict['MF'] = br_midfielders
+    if br_forwards == []:
+        pass
+    else:
+        bra_dict['GK'] = br_forwards
+    
+    
+    bel_dict = {}
+    bel_dict['GK'] = be_goalkeepers
+    bel_dict['MF'] = be_midfielders
+    bel_dict['FW'] = be_forwards
+    
+    sko_dict = {}
+    sko_dict['GK'] = sk_goalkeepers
+    sko_dict['MF'] = sk_midfielders
+    sko_dict['FW'] = sk_forwards
+    
+    all_country_dict = {}
+    all_country_dict['Argentina'] = arg_dict
+    all_country_dict['Brazil'] = bra_dict
+    all_country_dict['Belgium'] = bel_dict
+    all_country_dict['South Korea'] = sko_dict
+    
+    return all_country_dict
+'''  
+        if country == 'Argentina':
+            Argentina.append(arg2[i])
+            a_goalkeepers = []
+            a_midfielders = []
+            a_forwards = []
+            for i in range(len(arg2)):
+                player_pos = arg2[i].get('position')
+                if player_pos == 'GK':
+                    a_goalkeepers.append(arg2[i])
+                elif player_pos == 'MF':
+                    a_midfielders.append(arg2[i])
+                elif player_pos == 'FW':
+                    a_forwards.append(arg2[i]) 
+
+
+
+    arg_dict = {}
+    arg_dict['GK'] = a_goalkeepers
+    arg_dict['MF'] = a_midfielders
+    arg_dict['FW'] = a_forwards
+    
+    
+    all_country_dict = {}
+    all_country_dict['Argentina'] = arg_dict
+#    all_country_dict['Belgium'] = bel_dict
+    
+    return all_country_dict
+'''   
+
+
+
+'''   
+        
+        elif country == 'Brazil':
+            Brazil.append(arg1[i])
+            br_goalkeepers = []
+            br_midfielders = []
+            br_forwards = []
+            for i in range(len(arg1)):
+                player_pos = arg1[i].get('position')
+                if player_pos == 'GK':
+                    br_goalkeepers.append(arg1[i])
+                elif player_pos == 'MF':
+                    br_midfielders.append(arg1[i])
+                elif player_pos == 'FW':
+                    br_forwards.append(arg1[i])
+        elif country == 'South Korea':
+            South_Korea.append(arg1[i])
+    '''
+#    arg_dict = {}
+#    arg_dict['GK'] = a_goalkeepers
+#    arg_dict['MF'] = a_midfielders
+#    arg_dict['FW'] = a_forwards
+   
+#    bel_dict = {}
+#    bel_dict['GK'] = be_goalkeepers
+#    bel_dict['MF'] = be_midfielders
+#    bel_dict['FW'] = be_forwards
+    
+#    bra_dict = {}
+#    bra_dict['GK'] = br_goalkeepers
+#    bra_dict['MF'] = br_midfielders
+#    bra_dict['FW'] = br_forwards 
+    
+#    all_country_dict = {}
+#    all_country_dict['Argentina'] = arg_dict
+#    all_country_dict['Belgium'] = bel_dict
+#    all_country_dict['Brazil'] = br_forwards
+#    all_country_dict['South_Korea'] = South_Korea
+    
+#    return arg_dict
+
+
+pp = pprint.PrettyPrinter(depth=6)
+pp.pprint(players_by_country_and_position(SQUADS_DATA))
```

Comment:
This pprint line is what's causing your tests to fail. You commented out the import but you're still referencing it. Comment out this line and that should fix it.

Reply:
thank you!